src/ext/ca/serverca/scasched/scauser.cpp

```diff
@@ -515,7 +515,7 @@ HRESULT ScaUserExecute(
         if (wzDomain && *wzDomain)
         {
             er = ::DsGetDcNameW(NULL, wzDomain, NULL, NULL, NULL, &pDomainControllerInfo);
-            if (HRESULT_FROM_WIN32(er) == RPC_S_SERVER_UNAVAILABLE)
+            if (RPC_S_SERVER_UNAVAILABLE == er)
```

Comment:
Agree that this specific error code may not be the only one relevant to repeat the call with DS_FORCE_REDISCOVERY flag. I suggest doing it for any error code except ERROR_INVALID_DOMAINNAME, ERROR_NOT_ENOUGH_MEMORY